### ISSUE TITLE
Desktop-Safari: RC does not work properly in incognito/private mode

### ISSUE THREAD

**Issue by prinzdezibel** [reactioncommerce/reaction#1993]
Stock RC , running from localhost

Open incognito/private window in desktop Safari browser. Navigate to localhost:3000.

### Expected behavior
Reaction Commerce start page to be shown.

### Actual Behavior
Empty screen appears.

### Elusive solution
Hit refresh and everything seems fine. Except that it's not.

### Steps to reproduce
Put some items into cart and start Paypal-Express checkout. Ensure auth+capture is enabled in settings. After returning from PayPal's site (In my case the checkout process was NOT done in-context, but redirected to domain paypal.com. Be aware that this may differ depending on many things) an error is shown:

```
There was a problem with your payment.
ACK Failure: This transaction cannot be processed. The amount to be charged is zero.:10525
```

**Comment by prinzdezibel** [comment]
Apparently the implementation of private mode differs between Safari and other browsers. This could be the reason, why they behave differently after redirect from paypal.

https://gist.github.com/JohnAllsopp/1470989

**Comment by spencern** [comment]
I think this may have something to do with Safari not implementing LocalStorage in private mode.

We experience similar issues (though we don't use paypal) with GetOutfitted.

First load - safari private mode
https://www.dropbox.com/s/b19tibr025zo6i6/Screenshot%202017-03-15%2014.38.53.png?dl=0

Refresh and everything is fine.

This StackOverflow answer was helpful for me in understanding the safari private browser issue

http://stackoverflow.com/a/27081419/1179341

**Comment by spencern** [comment]
I was able to shim localStorage and prevent it from throwing errors, but the rendering issue persists, so it does not appear that it's related to the localStorage issue.

**Comment by prinzdezibel** [comment]
To have PayPal express checkout to work in Safari private mode, RC needs a way to re-login through preserving the current userLogin. 

I see several possibilities to achieve this:

1) Through setting a token/sessionId as cookie before navigating away to PayPal and read it again after PayPal has done its job and calls the redirect (`window.location.href = <shopUrl>/paypal/done`)

2) Setting a loginToken as CUSTOM field when calling `getExpressCheckoutToken` and pass it through the PayPal process. Retrieve this CUSTOM field after PayPal has done its job and calls the redirect (`window.location.href = <shopUrl>/paypal/done`) through PayPal's getExpressCheckoutDetails API and login via `loginWithToken`

3) Using the existing cookie `meteor_login_token`, wich should already be available as FastRender does set this. This cookie can probably used to checkin after PayPal has done it's job



Just for the records:  I used option 2 successfully in my project to enable checkout in Safari's private browsing mode.

**Comment by spencern** [comment]
I think this should be fixed by #2089

**Comment by zenweasel** [comment]
This issue is still happening.

**Comment by aaronjudd** [comment]
@zenweasel we need to dig into this one some more, I think we've squashed the other safari bugs, this is the last?

**Comment by abdulsemiu-atanda** [comment]
I'm able to make the payment using PayPal Express successful but there seems to be an issue when calling `cart/submitPayment` since it tries to copy the current cart to order. I'll take one more look at it today and raise a PR of possible

**Comment by aaronjudd** [comment]
@abdulsemiu-atanda did you make any progress on this?

**Comment by abdulsemiu-atanda** [comment]
I was blocked on `cart/submitPayment` as the method uses `Meteor.userId()` to fetch user cart which is usually a fresh cart in incognito/private mode of Safari.

**Comment by prinzdezibel** [comment]
Fixed: https://github.com/reactioncommerce/reaction/pull/2518

**Comment by zenweasel** [comment]
Fixed via #2518. Thanks again @prinzdezibel
